functions/dialect.go

```diff
@@ -176,18 +188,22 @@ func (d *dialectImpl) buildFunctionInfoMap(functions []dialectFunction) map[exte
 }
 
 type dialectFile struct {
-	Name               string                     `yaml:"name"`
-	Type               string                     `yaml:"type"`
-	SupportedTypes     map[string]dialectTypeInfo `yaml:"supported_types,omitempty"`
-	ScalarFunctions    []dialectFunction          `yaml:"scalar_functions,omitempty"`
-	AggregateFunctions []dialectFunction          `yaml:"aggregate_functions,omitempty"`
-	WindowFunctions    []dialectFunction          `yaml:"window_functions,omitempty"`
-	Dependencies       map[string]string          `yaml:"dependencies,omitempty"`
+	Name               string                      `yaml:"name"`
+	Type               string                      `yaml:"type"`
+	SupportedTypes     map[string]*dialectTypeInfo `yaml:"supported_types,omitempty"`
```

Comment:
why the change to pointer?

Reply:
Removed it. not needed after running SetDefaults on dialect file.